cli/gumball-machine/gumball-machine-cli.ts

```diff
@@ -0,0 +1,352 @@
+
+        const payerKeypair = loadWalletKey(payerKeypairPath);
+        const receiverPublicKey = new PublicKey(receiverPubkey);
+        const gumballMachinePublicKey = new PublicKey(gumballMachinePubkey);
+        const merkleRollPublicKey = new PublicKey(merkleRollPubkey);
+
+        const provider = await getProvider(url, payerKeypair);
+
+        const gumballMachine = new Program(GumballMachineIDL, GUMBALL_MACHINE_PROGRAM_ID, provider);
+
+        const dispenseNFTForSolIx =
+            await createDispenseNFTForSolIx(
+                dispenseNFTSolArgs,
+                payerKeypair.publicKey,
+                receiverPublicKey,
+                gumballMachinePublicKey,
+                merkleRollPublicKey,
+                GUMMYROLL_PROGRAM_ID,
+                BUBBLEGUM_PROGRAM_ID,
+                gumballMachine
+            );
+        const tx = new Transaction().add(dispenseNFTForSolIx);
+        await provider.send(tx, [payerKeypair], { commitment: "confirmed" });
+    });
+
+createCommand("dispense-token")
+    .description("Purchase compressed NFTs from a Gumball Machine with SPL Tokens")
+    .requiredOption(
+        "-t, --payer-tokens-pubkey <string>",
+        'Pubkey of the associated token account for the payers tokens'
+    )
+    .requiredOption(
+        "-r, --receiver-pubkey <string>",
+        'Pubkey of the fee receiver for the gumball machine'
+    )
+    .requiredOption(
+        "-g, --gumball-machine-pubkey <string>",
+        'Public key of the gumball machine'
+    )
+    .requiredOption(
+        "-m, --merkle-roll-pubkey <string>",
+        'Pubkey of the merkle roll account'
+    )
+    .action(async (options) => {
+        const { url, payerKeypairPath, receiverPubkey, gumballMachinePubkey, merkleRollPubkey, payerTokensPubkey } = options;
+
+        const payerKeypair = loadWalletKey(payerKeypairPath);
+        const payerTokensPublicKey = new PublicKey(payerTokensPubkey);
+        const receiverPublicKey = new PublicKey(receiverPubkey);
+        const gumballMachinePublicKey = new PublicKey(gumballMachinePubkey);
+        const merkleRollPublicKey = new PublicKey(merkleRollPubkey);
+
+        const provider = await getProvider(url, payerKeypair);
+
+        const gumballMachine = new Program(GumballMachineIDL, GUMBALL_MACHINE_PROGRAM_ID, provider);
+
+        const dispenseNFTForTokensIx =
+            await createDispenseNFTForTokensIx(
+                dispenseNFTTokenArgs,
+                payerKeypair.publicKey,
+                payerTokensPublicKey,
+                receiverPublicKey,
+                gumballMachinePublicKey,
+                merkleRollPublicKey,
+                GUMMYROLL_PROGRAM_ID,
+                BUBBLEGUM_PROGRAM_ID,
+                gumballMachine
+            );
+        const tx = new Transaction().add(dispenseNFTForTokensIx);
+        await provider.send(tx, [payerKeypair], { commitment: "confirmed" });
```

Comment:
Can we also log success/failure info about the tx after it is sent? I wrote related helper function to help do this (```async function execute``` ) somewhere in `contract/test/utils`.